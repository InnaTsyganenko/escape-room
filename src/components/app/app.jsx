import { ThemeProvider } from 'styled-components';
import {
  Switch,
  Route,
  BrowserRouter as Router,
} from 'components/common/common';
import DetailedQuest from 'components/detailed-quest/detailed-quest';
import Contacts from 'components/contacts/contacts';
import Home from 'components/home/home';
import NotFoundScreen from 'components/not-found-screen/not-found-screen';
import { appTheme } from './common';
import * as S from './app.styled';
import { AppRoute } from 'const';

const App = () => (
  <ThemeProvider theme={appTheme}>
    <S.GlobalStyle />
    <Router>
      <Switch>
        <Route exact path={`${AppRoute.QUEST}1`}>
          <DetailedQuest />
        </Route>
        <Route exact path={AppRoute.CONTACTS}>
          <Contacts />
        </Route>
        <Route exact path={AppRoute.ROOT}>
          <Home />
        </Route>
        <Route >
          <NotFoundScreen />
        </Route>
      </Switch>
    </Router>
  </ThemeProvider>
);

export default App;
