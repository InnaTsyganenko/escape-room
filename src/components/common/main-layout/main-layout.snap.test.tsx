import {render} from '@testing-library/react';
import { BrowserRouter } from 'components/common/common';
import MainLayout from './main-layout';
import { ThemeProvider } from 'styled-components';
import { appTheme } from '../../app/common';
import 'jest-styled-components';

describe('Component: MainLayout', () => {
  it('should render correctly', () => {
    const {container} = render(
        <ThemeProvider theme={appTheme}>
          <BrowserRouter history>
            <MainLayout children />
          </BrowserRouter>
        </ThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
