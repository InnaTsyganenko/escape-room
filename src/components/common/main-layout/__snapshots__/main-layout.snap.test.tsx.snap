// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MainLayout should render correctly 1`] = `
.c1 {
  max-width: 1366px;
  width: 100%;
  margin: 0 auto;
}

.c0 {
  z-index: 5;
  position: fixed;
  top: 0;
  left: 50%;
  width: 100vw;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  background-color: rgba(26,26,26,0.9);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  min-height: 74px;
  padding-top: 23px;
  padding-right: 33px;
  padding-bottom: 23px;
  padding-left: 32px;
}

.c3 {
  margin-right: 250px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  max-width: 600px;
  margin: 0;
  padding: 0;
  padding-top: 17px;
  padding-bottom: 17px;
  margin-bottom: -10px;
  list-style: none;
}

.c5:not(:last-child) {
  margin-right: 49px;
  margin-bottom: 10px;
}

.c6 {
  display: block;
  max-width: 100px;
  font-size: 14px;
  line-height: 16px;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  font-weight: 600;
  text-transform: uppercase;
  color: #F0F0F0;
  color: #F2890F;
}

.c6:focus,
.c6:hover {
  color: #F2890F;
}

.c7 {
  display: block;
  max-width: 100px;
  font-size: 14px;
  line-height: 16px;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  font-weight: 600;
  text-transform: uppercase;
  color: #F0F0F0;
}

.c7:focus,
.c7:hover {
  color: #F2890F;
}

.c8 {
  margin-top: 17px;
  margin-left: auto;
  font-size: 14px;
  line-height: 16px;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  font-weight: 600;
  color: #F0F0F0;
  font-feature-settings: 'pnum' on,'lnum' on;
}

.c8:focus,
.c8:hover {
  color: #F2890F;
}

@media (max-width:1300px) {
  .c3 {
    margin-right: auto;
  }
}

<header
  class="c0"
>
  <div
    class="c1 c2"
  >
    <a
      class="c3"
    >
      <img
        alt="Логотип Escape Room"
        class=""
        height="50"
        src="logo.svg"
        width="134"
      />
    </a>
    <nav
      class=""
    >
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          <a
            class="c6"
            href="/"
          >
            Квесты
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c7"
            href="/"
          >
            Новичкам
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c7"
            href="/"
          >
            Отзывы
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c7"
            href="/"
          >
            Акции
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c7"
            href="/contacts"
          >
            Контакты
          </a>
        </li>
      </ul>
    </nav>
    <a
      class="c8"
      href="tel:88003335599"
    >
      8 (800) 333-55-99
    </a>
  </div>
</header>
`;
